src/openforms/logging/logevent.py

```diff
@@ -40,10 +44,18 @@ def _create_log(
     if error:
         extra_data["error"] = str(error)
 
+    extra_data["avg"] = tag_avg
+
+    if user and not isinstance(user, User):
+        # If user is not of correct type (eg. AnonymousUser) we can not
+        #   save it on the TimelineLogProxy model
+        user = None
```

Comment:
I'm not sure if there is a better way to do this.  The `user` we save on the TimelineLog model needs to be what we use for `settings.AUTH_USER_MODEL`.  The current approach has two issues:

1. If we can the model we use for `settings.AUTH_USER_MODEL` this would break, though I think it's unlikely we'll change this setting.
2. It means we have to import `User` twice.  Once for the type checking and once in the function.  If we try importing it outside of the `if TYPE_CHECKING` then we get the `"AppRegistryNotReady: Apps aren't loaded yet."` exception and if we don't import it within the function then if type checking is turned off this code would break since the model would not be imported at all.

I'm happy to hear possible suggestions about how to improve this. :)

Reply:
You can use `User = django.contrib.auth.get_user_model()` to get the model that was configured for the User model.

However, there's no need for this I think. Simply do: `if user.is_authenticated`. See: https://docs.djangoproject.com/en/3.2/ref/contrib/auth/#attributes